src/content/docs/en/reference/experimental-flags/fonts.mdx

```diff
@@ -0,0 +1,718 @@
+
+<Tabs>
+
+<TabItem label="Relative path">
+
+```js
+src: ["./src/assets/fonts/MyFont.woff2", "./src/assets/fonts/MyFont.woff"]
+```
+
+</TabItem>
+
+<TabItem label="URL">
+
+```js
+src: [new URL("./custom.ttf", import.meta.url)]
+```
+
+</TabItem>
+
+<TabItem label="Package import">
+
+```js
+src: ["my-package/SomeFont.ttf"]
+```
+
+</TabItem>
+
+</Tabs>
+
+You can also specify a [tech](https://developer.mozilla.org/en-US/docs/Web/CSS/@font-face/src#tech) by providing objects:
+
+```js
+src: [{ url:"./src/assets/fonts/MyFont.woff2", tech: "color-COLRv1" }]
+```
+
+#### Other properties
+
+The following options from remote font families are also available for local font families within variants:
+
+- [display](#display)
+- [unicodeRange](#unicoderange)
+- [stretch](#stretch)
+- [featureSettings](#featuresettings)
+- [variationSettings](#variationsettings)
+
+```js title="astro.config.mjs" {14}
+import { defineConfig } from "astro/config"
+
+export default defineConfig({
+    experimental: {
+        fonts: [{
+            provider: "local",
+            name: "Custom",
+            cssVariable: "--font-custom",
+            variants: [
+                {
+                    weight: 400,
+                    style: "normal",
+                    src: ["./src/assets/fonts/custom-400.woff2"],
+                    display: "block",
+                },
+            ]
+        }]
+    }
+})
+```
+
+## Build your own font provider
+
+If you do not wish to use one of the [built-in providers](#available-remote-font-providers) (eg. you want to use a 3rd-party unifont provider or build something for a private registry), you can build your own.
+
+An Astro font provider is made up of two parts: the config object and the actual implementation.
+
+<Steps>
+
+1. Using the `defineAstroFontProvider()` type helper, create a function that returns a font provider config object containing:
+
+    - `entrypoint`: A URL, a path relative to the root or a package import.
+    - `config`: An optional serializable object passed to the unifont provider.
+
+    <Tabs>
+
+    <TabItem label="Without config">
+
+    ```ts title="provider/config.ts"
+    import { defineAstroFontProvider } from 'astro/config'
+
+    export function myProvider() {
+        return defineAstroFontProvider({
+            entrypoint: new URL('./implementation.js', import.meta.url),
+        })
+    }
+    ```
+
+    </TabItem>
+
+    <TabItem label="With config">
+
+    ```ts title="provider/config.ts"
+    import { defineAstroFontProvider } from 'astro/config'
+
+    interface Config {
+        // ...
+    }
+
+    export function myProvider(config: Config) {
+        return defineAstroFontProvider({
+            entrypoint: new URL('./implementation.js', import.meta.url),
+            config
+        })
+    }
+    ```
+
+    </TabItem>
+
+    </Tabs>
+
+2. Create a second file to export your unifont `provider` implementation:
+
+    ```ts title="myProvider.ts"
```

Comment:
``````suggestion
    ```ts title="implementation.ts"
``````
To be consistent with the entrypoint example above

Reply:
Thank you, I was trying to find out what the proper filename would be from the example and just guessed it was this!